tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -304,6 +304,117 @@ INSTANTIATE_TEST_SUITE_P(
         ::testing::Values(true),
         ::testing::Bool()));
 
+enum class SchedulingMode {
+  InterDeviceOnly,
+  Manual,
+};
+
+std::ostream& operator<<(std::ostream& out, const SchedulingMode& mode) {
+  switch (mode) {
+    case SchedulingMode::InterDeviceOnly:
+      return out << "InterDeviceOnly";
+    case SchedulingMode::Manual:
+      return out << "Manual";
+    default:
+      NVF_ERROR(false);
+  }
+  return out;
+}
+class PipelineTestStagedReduction
+    : public PipelineTest,
+      public ::testing::WithParamInterface<SchedulingMode> {};
+
+// 1D staged reduction
+// Inputs: X[A,B,C]
+TEST_P(PipelineTestStagedReduction, staged_reduction) {
```

Comment:
Nit: style convention. 

```suggestion
TEST_P(PipelineTestStagedReduction, StagedReduction) {
```